### DISCUSSION TITLE
Is there a way to disable Scroll Restoration ?

### DISCUSSION THREAD

**Discussion by FinPlorer** [TanStack/query#1977] [Q&A] [answered]
Hi everyone,

I'm using useInfiniteQuery to handle an infinite scroll list. When I change the key to update the list with differents data,  it goes directly to the last view page due to scroll restoration.

Is there a way to disable this ?

**Comment by TkDodo** [comment, marked as answer]
react-query doesn't actively do something with scroll restoration - the browser does that :) If you change the key and have no data, you will be in loading state. If you change the key and already have data, the data from the cache will be returned. if you are on page 3, and you change the key, and you also get 3 pages for the new key from the cache, your scroll position will stay where it is. Is that what you mean? If so, I think you'd need a `useEffect` that scrolls to the top whenever the key changes
